packages/core/src/components/Table/TableVirtualizedBody/TableVirtualizedBody.tsx

```diff
@@ -61,30 +74,42 @@ const TableVirtualizedBody = forwardRef(
         scrollOffset: number;
         scrollUpdateWasRequested: boolean;
       }) => {
+        if (virtualizedWithHeader) return;
         onScroll?.(scrollDirection, scrollOffset, scrollUpdateWasRequested);
       },
-      [onScroll]
+      [onScroll, virtualizedWithHeader]
     );
 
     const itemRenderer = useCallback<ComponentType<ListChildComponentProps<TableVirtualizedRow>>>(
       ({ index, style: { width: _width, ...style } }) => {
-        const currentItem = items[index];
+        if (virtualizedWithHeader && index === 0) {
+          return null; //placeholder for virtualized with header
+        }
+        const currentIndex = virtualizedWithHeader ? index - 1 : index;
+        const currentItem = items[currentIndex];
```

Comment:
```suggestion
        if (virtualizedWithHeader && index === items.length) {
          return null; //placeholder for virtualized with header
        }
        const currentItem = items[index];
```

would this work as well? or do you have to make the header as the index 0? (I know it makes sense it's index 0, but is `innerElementType` just puts it before the first index anyway?)

Reply:
`innerElementType` puts it before the first index but if we don't keep an empty slot for it on index 0 it overlaps the first row.

Reply:
got ya